app/Models/UserQuery.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/**
+ * Contains the description of a user query
+ * 
+ * It allows to extract the meaningful bits of the query to be manipulated in an
+ * easy way.
+ */
+class FreshRSS_UserQuery {
+
+	private $deprecated = false;
+	private $get;
+	private $get_name;
+	private $get_type;
+	private $name;
+	private $order;
+	private $search;
+	private $state;
+	private $url;
+	private $feed_dao;
+	private $category_dao;
+
+	/**
+	 * @param array $query
+	 * @param FreshRSS_Searchable $feed_dao
+	 * @param FreshRSS_Searchable $category_dao
+	 */
+	public function __construct($query, FreshRSS_Searchable $feed_dao = null, FreshRSS_Searchable $category_dao = null) {
+		$this->category_dao = $category_dao;
+		$this->feed_dao = $feed_dao;
+		if (isset($query['get'])) {
+			$this->parseGet($query['get']);
+		}
+		if (isset($query['name'])) {
+			$this->name = $query['name'];
+		}
+		if (isset($query['order'])) {
+			$this->order = $query['order'];
+		}
+		if (!isset($query['search'])) {
+			$query['search'] = '';
+		}
+		// linked to deeply with the search object, need to use dependency injection
```

Comment:
I am not sure to understand your question. Do you mean for the search object?
If your question is related to the search object, yes and no. I want the constructor not be dependent from that object but still want to use it. I want to use a container that holds the search object as well as DAO. I think the perfect location for that is in the context object.
This way we can remove lots of dependencies by using DAO from the context instead of recreating one every time we need one.
The new constructor signature would be `__construct($query, FreshRSS_Context $context)`